client/DAW/components/Timeline.js

```diff
@@ -44,15 +45,18 @@ class Timeline extends React.Component {
 
   componentDidMount() {
     // calling createSoundClips here for testing purposes, but will need to be done after project files array is retrieved
-    const { setFiles, setFilesThunk, addFileThunk, setClips, setClipsThunk, setTracks, setTracksThunk, setTempo, setTempoThunk, createSoundClips, clips, projectId, files, soundClips, addClipThunk, selectedTracks, length, setLengthThunk, setLength, tempo } = this.props;
+    const { setFiles, setFilesThunk, addFileThunk, setClips, setClipsThunk, setTracks, setTracksThunk, setTempo, setTempoThunk, createSoundClips, setWaveform, clips, projectId, files, soundClips, addClipThunk, selectedTracks, length, setLengthThunk, setLength, tempo } = this.props;
 
     // subscribe redux to firebase
     this.filesRef.on('value', snapshot => {
       const received = snapshot.val();
       setFiles(Object.assign({}, received));
       // createSoundClips checks for new files, gets them, and puts the audio buffer in the soundClips object
       console.log('received files are', snapshot.val());
-      createSoundClips(received, soundClips);
+      createSoundClips(received, soundClips)
+        .then(buffers => buffers.forEach(([id, audio]) => {
+          setWaveform(id, createWaveform(audio));
+        }));
```

Comment:
Took some hassle to get the right data to pass to the waveform builder function. Probably could have been put in the sound clip creation thunk, but hindsight...